readthedocs/core/views/__init__.py

```diff
@@ -124,7 +129,36 @@ def server_error_404(request, exception=None, template_name='404.html'):  # pyli
             )
         else:
             return response
-    r = render(request, template_name)
+
+    project, path = project_and_path_from_request(request, request.get_full_path())
+    version_slug = project.get_default_version()
```

Comment:
This fails when a 404 is generated from RTD itsefl because `project` is `None`, not a subdomain: http://localhost:8000/pagenotfound

Reply:
I'm thinking on writing a special `serve_error_404` for subdomains only and use it only on [`urls/subdomain.py`](https://github.com/bansalnitish/readthedocs.org/blob/c6ed7520932620fb7c4d200a72ce54b0255943c7/readthedocs/core/urls/subdomain.py#L24)

Any other idea?

Reply:
Yea, we shouldn't replace the default 404 for RTD, just for subdomains.